src/client/sky.cpp

```diff
@@ -640,113 +620,218 @@ void Sky::draw_sun(video::IVideoDriver *driver, float sunsize, const video::SCol
 		driver->drawIndexedTriangleFan(&vertices[0], 4, indices, 2);
 	}
 }
 
 
 void Sky::draw_moon(video::IVideoDriver *driver, float moonsize, const video::SColor &mooncolor,
 	const video::SColor &mooncolor2, float wicked_time_of_day)
-	/*
-	 * Draw moon in the sky.
-	 * driver: Video driver object used to draw
-	 * moonsize: the default size of the moon
-	 * mooncolor: main moon color
-	 * mooncolor2: second moon color
-	 * wicked_time_of_day: current time of day, to know where should be the moon in the sky
-	 */
+/*
+	* Draw moon in the sky.
+	* driver: Video driver object used to draw
+	* moonsize: the default size of the moon
+	* mooncolor: main moon color
+	* mooncolor2: second moon color
+	* wicked_time_of_day: current time of day, to know where should be the moon in
+	* the sky
+	*/
 {
 	static const u16 indices[4] = {0, 1, 2, 3};
 	std::array<video::S3DVertex, 4> vertices;
 	if (!m_moon_texture) {
 		driver->setMaterial(m_materials[1]);
 		const float moonsizes_1[4] = {
-				-moonsize * 1.9f,
-				-moonsize * 1.3f,
-				-moonsize,
-				-moonsize
-			};
+			(-moonsize * 1.9f) * m_moon_scale,
+			(-moonsize * 1.3f) * m_moon_scale,
+			(-moonsize) * m_moon_scale,
+			(-moonsize) * m_moon_scale
+		};
 		const float moonsizes_2[4] = {
-				moonsize * 1.9f,
-				moonsize * 1.3f,
-				moonsize,
-				moonsize * 0.6f
-			};
+			(moonsize * 1.9f) * m_moon_scale,
+			(moonsize * 1.3f) * m_moon_scale,
+			(moonsize) * m_moon_scale,
+			(moonsize * 0.6f) * m_moon_scale
+		};
 		video::SColor c1 = mooncolor;
 		video::SColor c2 = mooncolor;
 		c1.setAlpha(0.05 * 255);
 		c2.setAlpha(0.15 * 255);
 		const video::SColor colors[4] = {c1, c2, mooncolor, mooncolor2};
 		for (int i = 0; i < 4; i++) {
 			draw_sky_body(vertices, moonsizes_1[i], moonsizes_2[i], colors[i]);
-			place_sky_body(vertices, -90, wicked_time_of_day * 360 - 90);
+			place_sky_body(vertices, m_moon_rotation, wicked_time_of_day * 360 - 90);
 			driver->drawIndexedTriangleFan(&vertices[0], 4, indices, 2);
 		}
 	} else {
 		driver->setMaterial(m_materials[4]);
-		float d = moonsize * 1.9;
+		float d = (moonsize * 1.9) * m_moon_scale;
 		video::SColor c;
 		if (m_moon_tonemap)
 			c = video::SColor(0, 0, 0, 0);
 		else
 			c = video::SColor(255, 255, 255, 255);
 		draw_sky_body(vertices, -d, d, c);
-		place_sky_body(vertices, -90, wicked_time_of_day * 360 - 90);
+		place_sky_body(vertices, m_moon_rotation, wicked_time_of_day * 360 - 90);
 		driver->drawIndexedTriangleFan(&vertices[0], 4, indices, 2);
 	}
 }
 
+	void Sky::draw_stars(video::IVideoDriver * driver, float wicked_time_of_day)
+	{
+			driver->setMaterial(m_materials[1]);
+			// Tune values so that stars first appear just after the sun
+			// disappears over the horizon, and disappear just before the sun
+			// appears over the horizon.
+			// Also tune so that stars are at full brightness from time 20000
+			// to time 4000.
+			float starbrightness = MYMAX(0,	MYMIN(1, (0.25 - fabs(wicked_time_of_day < 0.5 ?
```

Comment:
this could use some reformatting
```cpp
float tod = wicked_time_of_day < 0.5f ? wicked_time_of_day : (1.0f - wicked_time_of_day);
float starbrightness = clamp((0.25f - fabs(tod)) * 20.0f, 0.0f, 1.0f);
```